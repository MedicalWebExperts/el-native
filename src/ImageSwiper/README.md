Image Swiper Component

This component is provided to give a swiper based on images. It gots autoplay by default
but can be changed.

It expects:

images (required): Array of images to generate the Swiper.
style: Extra styles to override the ones implemented. The component comes fully
styled, but can be changed.

IMPORTANT:
This component is based on 'react-native-swiper'. That means you have to install
it as a dependency in your project in order to work.

For more information head to https://github.com/leecade/react-native-swiper
