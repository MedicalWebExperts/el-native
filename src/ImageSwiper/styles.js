import { Dimensions } from 'react-native';

const { width } = Dimensions.get('window');

const palette = {
  darkTextColor: '#666666',
  opacity: 'rgba(0,0,0,0.2)',
};

const styles = {
  wrapper: {
  },
  slide: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: 'transparent',
  },
  image: {
    width,
    flex: 1,
  },
  dot: {
    backgroundColor: palette.opacity,
    width: 10,
    height: 10,
    borderRadius: 5,
    margin: 3,
  },
  activeDot: {
    backgroundColor: palette.darkTextColor,
    width: 10,
    height: 10,
    borderRadius: 5,
    margin: 3,
  },
  paginationStyle: {
    bottom: 8,
    left: null,
    right: 14,
  },
};

export default styles;
