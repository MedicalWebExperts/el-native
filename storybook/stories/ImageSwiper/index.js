import React from 'react';

import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import ImageSwiper from '../../../src/ImageSwiper';

storiesOf('ImageSwiper', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('default', () => <ImageSwiper images={[{ uri: 'https://facebook.github.io/react/img/logo_og.png' }, { uri: 'https://facebook.github.io/react/img/logo_og.png' }]} />);
