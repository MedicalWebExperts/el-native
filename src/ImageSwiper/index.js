import React, { PropTypes } from 'react';
import { View, Image, StyleSheet } from 'react-native';
import Swiper from 'react-native-swiper';

import styles from './styles';

const proptypes = {
  images: PropTypes.array.isRequired,
  style: PropTypes.object,
};

const defaultProps = {
  images: [],
  styles: {},
};

function ImageSwiper({ images, style }) {
  const stylesCommon = style ? StyleSheet.flatten([styles, style]) : styles;
  const swiper = (
    <View>
      <Swiper
        style={stylesCommon.wrapper} height={240}
        dot={<View style={stylesCommon.dot} />}
        activeDot={<View style={stylesCommon.activeDot} />}
        paginationStyle={stylesCommon.paginationStyle}
        autoplay
      >
        {images.map(image =>
          <View style={stylesCommon.slide}>
            <Image
              resizeMode="stretch"
              style={stylesCommon.image}
              source={image}
            />
          </View>)
        }
      </Swiper>
    </View>
  );
  return swiper;
}

ImageSwiper.proptypes = proptypes;
ImageSwiper.defaultProps = defaultProps;

export default ImageSwiper;
