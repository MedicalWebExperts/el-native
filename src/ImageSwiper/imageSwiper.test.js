import React from 'react';
import renderer from 'react-test-renderer';

import ImageSwiper from './index';

const images = [
  { uri: 'https://facebook.github.io/react/img/logo_og.png' },
  { uri: 'https://facebook.github.io/react/img/logo_og.png' },
];

describe('Snapshots', () => {
  it('should render default image swiper', () => {
    const component = renderer.create(<ImageSwiper images={images} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
