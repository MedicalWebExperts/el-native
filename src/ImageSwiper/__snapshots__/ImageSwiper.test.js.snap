// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render default image swiper 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "position": "relative",
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      activeDot={
        <View
          style={
            Object {
              "backgroundColor": "#666666",
              "borderRadius": 5,
              "height": 10,
              "margin": 3,
              "width": 10,
            }
          }
        />
      }
      automaticallyAdjustContentInsets={false}
      autoplay={true}
      autoplayDirection={true}
      autoplayTimeout={2.5}
      bounces={false}
      contentContainerStyle={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {},
        ]
      }
      contentOffset={
        Object {
          "x": 0,
        }
      }
      disableNextButton={false}
      dot={
        <View
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.2)",
              "borderRadius": 5,
              "height": 10,
              "margin": 3,
              "width": 10,
            }
          }
        />
      }
      height={240}
      horizontal={true}
      index={0}
      loadMinimal={false}
      loadMinimalSize={1}
      loop={true}
      onIndexChanged={[Function]}
      onMomentumScrollEnd={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      paginationStyle={
        Object {
          "bottom": 8,
          "left": null,
          "right": 14,
        }
      }
      pagingEnabled={true}
      removeClippedSubviews={true}
      scrollsToTop={false}
      showsButtons={false}
      showsHorizontalScrollIndicator={false}
      showsPagination={true}
      showsVerticalScrollIndicator={false}
      style={undefined}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "height": 240,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMode="stretch"
              source={
                Object {
                  "uri": "https://facebook.github.io/react/img/logo_og.png",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "width": 750,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 240,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMode="stretch"
              source={
                Object {
                  "uri": "https://facebook.github.io/react/img/logo_og.png",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "width": 750,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 240,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMode="stretch"
              source={
                Object {
                  "uri": "https://facebook.github.io/react/img/logo_og.png",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "width": 750,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 240,
                "width": 750,
              },
              Object {
                "backgroundColor": "transparent",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Image
              resizeMode="stretch"
              source={
                Object {
                  "uri": "https://facebook.github.io/react/img/logo_og.png",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "width": 750,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "bottom": 25,
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
          },
          Object {
            "bottom": 8,
            "left": null,
            "right": 14,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#666666",
            "borderRadius": 5,
            "height": 10,
            "margin": 3,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.2)",
            "borderRadius": 5,
            "height": 10,
            "margin": 3,
            "width": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
